cmd/internal/pull.go

```diff
@@ -71,7 +71,7 @@ func (cmd *PullCommand) Run(config *phrase.Config) error {
 		val, ok := localesCache[LocalesCacheKey{target.ProjectID, target.GetBranch()}]
 		if !ok || len(val) == 0 {
 			if cmd.Branch != "" {
-				continue
+				return fmt.Errorf("Branch '%s' does not exist in project '%s'", cmd.Branch, target.ProjectID)
```

Comment:
Changes have to be made within the [OpenAPI Repo](https://github.com/phrase/openapi/tree/master/clients/cli)